ginivision/src/main/java/net/gini/android/vision/internal/camera/photo/PhotoEdit.java

```diff
@@ -13,55 +13,50 @@
 public class PhotoEdit {
 
     private final Photo mPhoto;
-    private final List<PhotoModifier> mPhotoModifiers;
+    private List<PhotoModifier> mPhotoModifiers;
 
     PhotoEdit(@NonNull Photo photo) {
         mPhoto = photo;
-        mPhotoModifiers = new ArrayList<>();
+    }
+
+    private List<PhotoModifier> getPhotoModifiers() {
+        if (mPhotoModifiers == null) {
+            mPhotoModifiers = new ArrayList<>();
+        }
+        return mPhotoModifiers;
     }
 
     @NonNull
     public PhotoEdit rotateTo(int degrees) {
         final PhotoRotationModifier rotationModifier = new PhotoRotationModifier(degrees, mPhoto);
-        mPhotoModifiers.add(rotationModifier);
+        getPhotoModifiers().add(rotationModifier);
         return this;
     }
 
     @NonNull
     public PhotoEdit compressBy(int quality) {
         final PhotoCompressionModifier compressionModifier = new PhotoCompressionModifier(quality, mPhoto);
-        mPhotoModifiers.add(compressionModifier);
+        getPhotoModifiers().add(compressionModifier);
```

Comment:
Is it possible to check here that only one compression modifier is in that list? Would be more defensive.